### PR TITLE
[DPC-3984] Me/dpc 3984 handle empty response files

### PR DESCRIPTION
**Pull request by MEspositoE14s**
## 🎫 Ticket

https://jira.cms.gov/browse/DPC-3984

## 🛠 Changes

- Fix to handle response files with no rows.
- Fixes to `make test` and `make integration-test`.

## ℹ️ Context for reviewers

In testing with the NGD team, we realized that our lambda couldn't handle a response file with no rows.  This update makes sure that empty files are handled and marked as processed in the DB.

## ✅ Acceptance Validation

New unit test written, and all existing unit and integration tests are passing.

## 🔒 Security Implications

- [ ] This PR adds a new software dependency or dependencies.
- [ ] This PR modifies or invalidates one or more of our security controls.
- [ ] This PR stores or transmits data that was not stored or transmitted before.
- [ ] This PR requires additional review of its security implications for other reasons.

If any security implications apply, add Jason Ashbaugh (GitHub username: StewGoin) as a reviewer and do not merge this PR without his approval.

### CODE REVIEW

File: lambda/opt-out-import/synthetic_test_data/T.NGD.DPC.RSP.D010424.T1122001.IN [hunk 1]
```diff
@@ -0,0 +1,2 @@
+HDR_BENEDATARSP20240205
+TRL_BENEDATARSP202402050000000007
```
Comment (kyeah) [inline lambda/opt-out-import/synthetic_test_data/T.NGD.DPC.RSP.D010424.T1122001.IN:2]:
```suggestion
TRL_BENEDATARSP202402050000000000
```

Doesn't really matter but for realism 😄

**Review by kyeah — APPROVED** [review]

File: lambda/opt-out-import/db_test.go [hunk 2]
```diff
@@ -229,6 +229,61 @@ func TestInsertConsentRecords_DatabaseError(t *testing.T) {
 	assert.Equal(t, 0, len(response))
 }
 
+func TestInsertConsentRecordsEmptyFile(t *testing.T) {
```
Comment (kyeah) [inline lambda/opt-out-import/db_test.go:232]: Just a thought but I wonder if we might want to ensure that generating the confirmation file also works with an empty array/file passed into it.

File: lambda/opt-out-import/synthetic_test_data/T.NGD.DPC.RSP.D010424.T1122001.IN [hunk 3]
```diff
@@ -0,0 +1,2 @@
+HDR_BENEDATARSP20240205
```
Comment (kyeah) [inline lambda/opt-out-import/synthetic_test_data/T.NGD.DPC.RSP.D010424.T1122001.IN:1]: can we add this to `populate_s3.sh` as well? (sorry, adding non-blocking comments after I approved 🙈)

File: lambda/opt-out-import/db_test.go (hunk 2 above)
Comment (kyeah) [reply lambda/opt-out-import/db_test.go:232]: I did verify manually that it runs all the way through regardless 👍 

```
(venv) opt-out-import: make run-local
ENV="local" DB_USER_DPC_CONSENT="postgres" DB_PASS_DPC_CONSENT="dpc-safe" DB_HOST="localhost" IS_TESTING=true go run main.go parsers.go utils.go models.go db.go
Importing opt out file: bfdeft01/dpc/in/T.NGD.DPC.RSP.D010424.T1122001.IN (bucket: demo-bucket)INFO[0000] 95 <nil>                                     
INFO[0000] Successfully imported opt out file metadata. 
INFO[0000] file downloaded: size=57                     
INFO[0000] No consent records to insert.                
INFO[0000] Created consent records with the following ID fields:  created_records_count=0
INFO[0000] true   
```

Comment (MEspositoE14s) [reply lambda/opt-out-import/db_test.go:232]: No problem, that's actually a little a little easier to do!  I only did it the other way to keep it consistent with the other tests.

Comment (kyeah) [reply lambda/opt-out-import/db_test.go:232]: yeah definitely! I think two tests are helpful.

**Pull request merged**